contract_journal/models/contract.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    @api.model
+    def _default_journal(self):
+        company_id = self._context.get(
+            'company_id', self.env.user.company_id.id)
+        domain = [
+            ('type', '=', 'sale'),
+            ('company_id', '=', company_id)]
+        return self.env['account.journal'].search(domain, limit=1)
+
+    journal_id = fields.Many2one(
+        'account.journal',
+        string='Journal',
+        default=_default_journal,
+        domain="[('type', '=', 'sale'),('company_id', '=', company_id)]")
+
+    def _prepare_invoice_data(self, cr, uid, contract, context=None):
+        context = context or {}
+        invoice = super(AccountAnalyticAccount, self)._prepare_invoice_data(
```

Comment:
s/invoice/invoice_vals (to highlight that it's a dictionary with vals, not a recordset)